docs/proposals/22-autoscaling-kube-apiserver-via-independent-hpa-and-vpa.md

```diff
@@ -0,0 +1,354 @@
+    - [Existing Solution](#existing-solution)
+    - [Deficiencies of the Existing Solution](#deficiencies-of-the-existing-solution)
+      - [Algorithmic limitation: fusing the behavior of two blackbox systems](#algorithmic-limitation-fusing-the-behavior-of-two-blackbox-systems)
+      - [HVPA stuck in unintended stable equilibrium due to feedback of non-deterministic "replica count" signal](#hvpa-stuck-in-unintended-stable-equilibrium-due-to-feedback-of-non-deterministic-replica-count-signal)
+      - [Maintaining inactive code](#maintaining-inactive-code)
+      - [Inefficiency induced by initial horizontal-only scaling stage](#inefficiency-induced-by-initial-horizontal-only-scaling-stage)
+      - ['MinChange' stabilisation prevents response to node CPU exhaustion](#minchange-stabilisation-prevents-response-to-node-cpu-exhaustion)
+      - [HPA-VPA policy fusion model does not account for multiplicative interaction](#hpa-vpa-policy-fusion-model-does-not-account-for-multiplicative-interaction)
+    - [Other Benefits to Replacing the Existing Solution](#other-benefits-to-replacing-the-existing-solution)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Proposal](#proposal)
+    - [Rationale](#rationale)
+    - [Design Outline](#design-outline)
+    - [Element: Gardener Custom Metrics Provider Component](#element-gardener-custom-metrics-provider-component)
+    - [Element: New Custom Pod Metric for ShootKapis](#element-new-custom-pod-metric-for-shootkapis)
+    - [Element: HPA](#element-hpa)
+    - [Element: VPA](#element-vpa)
+    - [Transition Strategy](#transition-strategy)
+    - [Vision Beyond This Enhancement](#vision-beyond-this-enhancement)
+  - [Discussion and Limitations](#discussion-and-limitations)
+      - [Gardener-custom metrics precludes the use of other sources of custom metrics](#gardener-custom-metrics-precludes-the-use-of-other-sources-of-custom-metrics)
+      - [Observed fluctuations in actual compute efficiency affecting ShootKapi workloads](#observed-fluctuations-in-actual-compute-efficiency-affecting-shootkapi-workloads)
+  - [Alternatives](#alternatives)
+  - [References](#references)
+
+## Summary
+When it comes to autoscaling shoot control plane `kube-apiserver` instances (ShootKapi hereafter), Gardener needs
+both stability and efficiency (accurate scaling). The existing approach of fusing HPA and VPA into the 2-dimensional
+autoscaler which is HVPA, poses severe algorithmic limitations which manifest as stability and efficiency issues in the
+field. 
+
+This document outlines the main deficiencies in the existing solution and proposes a simple design which overcomes them,
+reduces maintenance effort by reducing Gardener's custom code base, and enables subsequent reduction in ShootKapi
+compute cost.
+
+## Motivation
+Gardener uses a relatively complex approach for scaling shoot control plane `kube-apiserver` instances (ShootKapi
+hereafter). This need is a consequence of the following concerns:
```

Comment:
Thanks! I intentionally duplicated the introduction, but didn't notice that it's annoying in that form (which it very much is).

I removed the "hereafter" from the second introduction. Now it reads just '(ShootKapi)' Hopefully that's a good compromise between the annoyance of a redundancy, and reinforcing the abbreviation in case it didn't stick in the reader's head the first time.